Fix: improve facility location handling in create/edit form

siva222003 opened the pull request:
## Proposed Changes

- Fixes #12226 
- Added a condition to check whether the latitude and longitude values are 0/0 before rendering

[Screencast from 2025-05-05 22-14-05.webm](https://github.com/user-attachments/assets/f5b7d3f1-0ef0-479b-b0c7-dd6db68cbb4d)


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
  - Improved validation for facility map links to prevent display of links with invalid or zero coordinates.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

siva222003 committed:
Fix: improve facility location handling in create/edit form

siva222003 commented:
@rithviknishad @Jacobjeevan
I’ve added a reset button to clear the selected location in both the create and edit facility forms. However, if a location has been added at least once, removing the selected location and submitting the form will not remove the location from the facility.

Please let me know if any changes are needed.

Jacobjeevan commented:
> @rithviknishad @Jacobjeevan I’ve added a reset button to clear the selected location in both the create and edit facility forms. However, if a location has been added at least once, removing the selected location and submitting the form will not remove the location from the facility.
> 
> Please let me know if any changes are needed.

If I recall correctly, BE doesn't allow us to overwrite lat/log to undefined (once it's filled, hence why we added the fallback value). So, what would be the use case of a reset button here? 🤔 

@nihal467 Instead, we can hide the show on map link if coords are 0/0, no? 🤔

siva222003 commented:
> > @rithviknishad @Jacobjeevan I’ve added a reset button to clear the selected location in both the create and edit facility forms. However, if a location has been added at least once, removing the selected location and submitting the form will not remove the location from the facility.
> > Please let me know if any changes are needed.
> 
> If I recall correctly, BE doesn't allow us to overwrite lat/log to undefined (once it's filled, hence why we added the fallback value). So, what would be the use case of a reset button here? 🤔
> 
> @nihal467 Instead, we can hide the show on map link if coords are 0/0, no? 🤔

@Jacobjeevan Yeah, we can't overwrite it once it's added. We can go with your approach as well, but the `reset` here is useful when you've filled the form, added the location, and want to clear it without closing the form and starting over.(not sure how significant that use case is, though)

nihal467 committed:
Merge branch 'develop' into issues/12226/facility-location-handling

nihal467 commented:
yes, we can hide them if its 0/0 , can you make the change @siva222003

siva222003 commented:
Okay @nihal467

siva222003 committed:
add condition for rendering location

siva222003 commented:
@rithviknishad @Jacobjeevan @nihal467 Made the changes.

nihal467 commented:
LGTM
